File: backend/app/models/resource.py
# backend/app/models/resource.py

from sqlalchemy import Column, String, Text, Enum, ForeignKey, DateTime, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
import uuid, enum
from app.db.base import Base

class ResourceType(str, enum.Enum):
    link     = "link"
    file     = "file"
    note     = "note"
    github   = "github"
    figma    = "figma"
    notion   = "notion"
    other    = "other"

class Resource(Base):
    __tablename__ = "resources"

    id           = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    hackathon_id = Column(UUID(as_uuid=True), ForeignKey("hackathons.id"), nullable=False)
    user_id      = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False)
    title        = Column(String, nullable=False)
    url          = Column(String, nullable=True)
    description  = Column(Text, nullable=True)
    type         = Column(Enum(ResourceType), default=ResourceType.link)
    created_at   = Column(DateTime(timezone=True), server_default=func.now())

    hackathon = relationship("Hackathon", back_populates="resources")
    user      = relationship("User", back_populates="resources")

File: frontend/types/resource.ts
// frontend/types/resource.ts

export type ResourceType = "link" | "file" | "note" | "github" | "figma" | "notion" | "other";

export interface Resource {
    id: string;
    hackathon_id: string;
    user_id: string;
    title: string;
    url: string | null;
    description: string | null;
    type: ResourceType;
    created_at: string;
}

export interface ResourceCreate {
    hackathon_id: string;
    title: string;
    url?: string;
    description?: string;
    type?: ResourceType;
}